src/components/QrcodeStream.vue

```diff
@@ -6,279 +6,283 @@
-
-          const xScalar = uncutWidth / resolutionWidth;
-          const yScalar = uncutHeight / resolutionHeight;
-          const xOffset = (displayWidth - uncutWidth) / 2;
-          const yOffset = (displayHeight - uncutHeight) / 2;
-
-          const scale = ({ x, y }) => {
-            return {
-              x: Math.floor(x * xScalar),
-              y: Math.floor(y * yScalar)
-            };
-          }
-
-          const translate = ({ x, y }) => {
-            return {
-              x: Math.floor(x + xOffset),
-              y: Math.floor(y + yOffset)
-            };
-          }
-
-          const adjustedCodes = detectedCodes.map(detectedCode => {
-            const { boundingBox, cornerPoints } = detectedCode
-
-            const { x, y } = translate(scale({
-              x: boundingBox.x,
-              y: boundingBox.y
-            }))
-            const { x: width, y: height } = scale({
-              x: boundingBox.width,
-              y: boundingBox.height
-            })
-
-            return {
-              ...detectedCode,
-              cornerPoints: cornerPoints.map(point => translate(scale(point))),
-              boundingBox: DOMRectReadOnly.fromRect({ x, y, width, height })
-            }
-          })
-
-          canvas.width = video.offsetWidth;
-          canvas.height = video.offsetHeight;
-
-          const ctx = canvas.getContext('2d');
 
-          this.track(adjustedCodes, ctx);
-        } else {
-          this.clearCanvas(canvas);
+      const translate = ({ x, y }: Point) => {
+        return {
+          x: Math.floor(x + xOffset),
+          y: Math.floor(y + yOffset)
         }
       }
-    },
 
-    repaintTrackingLayer(video, canvas, location) {
-      const ctx = canvas.getContext("2d");
+      const adjustedCodes = detectedCodes.map((detectedCode) => {
+        const { boundingBox, cornerPoints } = detectedCode
 
+        const { x, y } = translate(
+          scale({
+            x: boundingBox.x,
+            y: boundingBox.y
+          })
+        )
+        const { x: width, y: height } = scale({
+          x: boundingBox.width,
+          y: boundingBox.height
+        })
+
+        return {
+          ...detectedCode,
+          cornerPoints: cornerPoints.map((point) => translate(scale(point))),
+          boundingBox: DOMRectReadOnly.fromRect({ x, y, width, height })
+        }
+      })
 
-      window.requestAnimationFrame(() => {
-        canvas.width = displayWidth;
-        canvas.height = displayHeight;
+      canvas.width = video.offsetWidth
+      canvas.height = video.offsetHeight
 
-        this.trackRepaintFunction(coordinatesAdjusted, ctx);
-      });
-    },
+      const ctx = canvas.getContext('2d')
 
-    clearCanvas(canvas) {
-      const ctx = canvas.getContext("2d");
-      ctx.clearRect(0, 0, canvas.width, canvas.height);
+      props.track(adjustedCodes, ctx)
+    } else {
+      clearCanvas(canvas)
     }
-
   }
-};
```

Comment:
Ok, there are so many changes in this file. It's hard for me to keep track of what actually has changed. Especially the `QrcodeStream.vue` is a bit sensitive to changes. I'm sure it's a good idea to migrate to Composition API. But I would be grateful if this can be done in a second step.

Reply:
:0 I worked on that based on that you said

> Only acceptance criterium: I can view and test the deployed demo page without doing any setup on my machine.

https://github.com/gruhn/vue-qrcode-reader/issues/203#issuecomment-1431324053

Now I put much effort into this and also typescript being included now really makes it hard to make mistakes.

Reply:
I know, sorry. Typescript is great but not a silver bullet. The thing is `QrcodeStream` is very fragile. A lot of stuff is happening async and the order is important. It's also particularly easy to break responsiveness. Ok, but are you saying you really changed nothing regarding the components function?

Reply:
I think all documented demos work as expected, so I don't know what else I could ensure. Did you find any issues?

Reply:
I mean I also get why there is a higher likelyhood that some changes break something here, but pulling it out here and creating another PR for it would have us looking at the same diff, wouldn't it?